### PR TITLE
SPII-32: Joystick controller

### PR DESCRIPTION
**Pull request by aniket11bh**
### Changes
- Add new package `vrx_control`
- Integrate joystick controller
- Change the max speed of the bot to `5 knots`

### Extra install
```bash
sudo apt install ros-humble-joy-teleop
```

### Test:
1. Launch simulator (gazebo)
2. Launch rviz and localization package using consolidated launch file `ros2 launch vrx_launch vrx.launch.py`
3. Launch joystick controller `ros2 launch vrx_control joy_teleop.py`
4. Drive the robot using joystick

## Joystick mapping

https://github.com/turtlebot/turtlebot4_robot/blob/galactic/turtlebot4_bringup/config/turtlebot4_controller.config.yaml

Ensure that the gamepad is in D config, i.e, 6 axis

- The `L1` button needs to be pressed all the time in addition to the following commands (dead man's switch),
- the right stick up/down axis (axis 3) is mapped to the thruster forward/reverse,
- the left stick left/right axis (axis 0) is mapped to rotating the thrusters clockwise/counter-clockwise

### Verify
Verify that at the max. thrust, the robot's velocity is near around `5 knots` or `2.54 m/s`
Check robot's velocity on topic : `/odometry/filtered/twist/twist/linear/x`

joy-drive-velocity

### CODE REVIEW

File: .vscode/c_cpp_properties.json
Comment (sumandeepb) [inline .vscode/c_cpp_properties.json:1]: this has local settings, should not be checked in.

File: .vscode/settings.json
Comment (sumandeepb) [inline .vscode/settings.json:1]: this has local settings, should not be checked in.

**Review by sumandeepb — CHANGES_REQUESTED** [review]
Add to .gitignore:

.vscode/

If there are specific debug and/or editor settings that needs to be standardized, we can do that seperately using specifig settings files.

**Comment by aniket11bh** [comment]
@sumandeepb can you approve this?

**Review by sumandeepb — APPROVED** [review]

**Pull request merged**